platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java

```diff
@@ -851,14 +799,91 @@ public void deserialize(SerializableDataInputStream inStream, int version) throw
     /**
      * Return the String of dot format of the IPv4 address.
      *
-     * @param ip
-     * 		IP address.
+     * @param ip IP address.
      * @return IP address String of dot format.
      */
     public static String ipString(byte[] ip) {
         return "" + (0xff & ip[0]) + "." + (0xff & ip[1]) + "." + (0xff & ip[2]) + "." + (0xff & ip[3]);
     }
 
+    /**
+     * Translates this address to the string format used in config.txt.
+     *
+     * @return the string form of this address that would appear in config.txt.
+     */
+    public String toConfigText() {
+        return "address, " + nickname + ", " + selfName + ", " + stake + ", " + ipString(addressInternalIpv4) + ", "
+                + portInternalIpv4 + ", " + ipString(addressExternalIpv4) + ", " + portExternalIpv4;
+    }
+
+    /**
+     * Parses an address from a single line of text.  The address must be in the form produced by
+     * {@link #toConfigText()}
+     *
+     * @param addressLine the string to parse an Address form.
+     * @param id the id to give to the parsed Address.
+     * @param isOwnHostDeterminer a function to determine if isOwn should be true given an InetAddress.
+     * @param memo the memo text for the address.
+     * @return the Address parsed from the addressLine.
+     * @throws ParseException if there is any problem with creating an Address from the addressLine.
+     */
+    public static Address parseConfigText(
```

Comment:
Its better if this is in a separate utility class

Reply:
moved to AddressBookUtils.java